azurerm/internal/services/network/express_route_connection_resource.go

```diff
@@ -163,11 +160,14 @@ func resourceExpressRouteConnectionCreateUpdate(d *pluginsdk.ResourceData, meta
 				ID: utils.String(d.Get("express_route_circuit_peering_id").(string)),
 			},
 			EnableInternetSecurity: utils.Bool(d.Get("enable_internet_security").(bool)),
-			RoutingConfiguration:   expandExpressRouteConnectionRouting(d.Get("routing").([]interface{})),
 			RoutingWeight:          utils.Int32(int32(d.Get("routing_weight").(int))),
 		},
 	}
 
+	if v, ok := d.GetOk("routing"); ok {
+		parameters.ExpressRouteConnectionProperties.RoutingConfiguration = expandExpressRouteConnectionRouting(v.([]interface{}))
+	}
+
```

Comment:
Wondering why this check is needed here?

Reply:
It's optional property. So added a check.

Reply:
Isn't this already checked in the expand function?

Reply:
Updated